src/profile-logic/profile-data.js

```diff
@@ -2609,15 +2609,23 @@ export function extractProfileFilterPageData(
     return null;
   }
 
-  // Getting the first page's innerWindowID and then getting its url.
-  const innerWindowID = [...relevantPages][0];
-  const filteredPages = pages.filter(
-    page => page.innerWindowID === innerWindowID
+  // Getting the pages that are relevant and a topmost frame.
+  let filteredPages = pages.filter(
+    page =>
+      relevantPages.has(page.innerWindowID) && page.embedderInnerWindowID === 0
   );
 
-  if (filteredPages.length !== 1) {
-    // There should be only one page with the given innerWindowID, they are unique.
-    console.error(`Expected one page but ${filteredPages.length} found.`);
+  if (filteredPages.length > 1) {
+    // If there are more than one topmost page, it's also good to filter out the
+    // `about:` pages so user can see their url they are actually profiling.
+    filteredPages = [
+      filteredPages.filter(page => !page.url.startsWith('about:'))[0],
+    ];
+  }
+
+  if (filteredPages.length === 0) {
+    // There should be at least one relevant page.
+    console.error(`Expected a relevant page but couldn't find it.`);
     return null;
   }
 
```

Comment:
at this point, if all we have is an `about:*` page, the code below will throw (and we'll return null).

I think we should be more explicit with `about:*` pages, maybe with this right before the `try`:
```js
if (pageUrl.startsWith('about:')) {
    return {
      origin: pageUrl,
      hostname: pageUrl,
      favicon: null,
    };
}
```
Then we would have at least something like `about:home` written as title, which I think would be a nice thing.
I don't know if the rest of the code and the types support this, so I'm fine if you want to move that to a separate issue.

If we don't want this, then the alternative is removing `if (filteredPages.length > 1) {` above, which is useless in the current situation. (but keep it if you think my suggestion is valid but just can't do it now).

What do you think?

Reply:
Yeah, I think it's better to have a case for `about:*` pages. Updated to code to include it and added a test for that as well.